crates/ra_assists/src/flip_eq_operands.rs

```diff
@@ -0,0 +1,87 @@
+use hir::db::HirDatabase;
+use ra_syntax::{
+    ast::{AstNode, BinExpr, BinOp}
+};
+
+use crate::{AssistCtx, Assist, AssistId};
+
+pub(crate) fn flip_eq_operands(mut ctx: AssistCtx<impl HirDatabase>) -> Option<Assist> {
+    let expr = ctx.node_at_offset::<BinExpr>()?;
+    let allowed_ops = [BinOp::EqualityTest, BinOp::NegatedEqualityTest];
+    let expr_op = expr.op()?;
```

Comment:
Let’s also check hat the cursor is exactly on the operator? That is, that ctx.coffset is contained in op.range

Reply:
How can I get the exact position of the operator? Something like that would be fine?
```rust
let lhs_range = expr.lhs()?.syntax().range();
let rhs_range = expr.rhs()?.syntax().range();
let op_node_range = TextRange::from_to(lhs_range.end(), rhs_range.start());
if ctx.frange.range.is_subrange(&op_node_range) { ... }
```

Reply:
Good question, looks like there's no method to do this. Let's add it!

I think we should rename `BinExpr::op` and `PrefixExpr::op` to `op_kind`, and reuse `op`  method to return a `SyntaxNode`

Reply:
I can try, but for this use case wouldn't actually be better if the assist applied even when there is some white space? For example if the cursor is here: `1<|>   == 2` the assist should be available or not?

Reply:
The problem with assists is that there are usually many of them, so keeping focus zone precise helps in general